programs/drift/src/controller/orders.rs

```diff
@@ -1931,8 +1931,16 @@ fn fulfill_perp_order(
             .fuel_perp_delta(market_index, -maker_base_asset_amount_filled)
             .fuel_numerator(&maker, now);
 
-        if oracle_stale_for_fill && maker_position_increasing {
-            context = context.margin_ratio_override(MARGIN_PRECISION);
+        if oracle_stale_for_margin {
+            validate!(
+                user_order_position_decreasing || !maker_position_increasing,
+                ErrorCode::InvalidOracle,
+                "taker or maker must be reducing position if oracle stale for margin"
+            )?;
```

Comment:
another validate on market account's open interest before and after would be great (if doesn't involve too much variable passing)